### PR TITLE
Records are now passed as copies

### PR DESCRIPTION
**Pull request by oskarcieslik**
Implements #34

### CODE REVIEW

File: src/database.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export class Database {
 
   all(collectionName) {
     this.checkFactoryPresence(collectionName);
-    return this.store[collectionName];
+    return _.cloneDeep(this.store[collectionName]);
```
Comment (zzarcon) [inline src/database.js]: So, we only need to do it here since all the other accessor functions use `all`, right? 👍

**Comment by zzarcon** [comment]
Maybe out of scope, but what about providing `findById` method?

File: src/recordFactory.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+
+const updateRecord = (record, collectionName, store) => {
+  const originalRecord = _.find(store[collectionName], {id: record.id});
   return Object.assign(originalRecord, record);
 };
 
-export const recordFactory = (record, collectionName, database) => {
-  record.save = () => updateRecord(collectionName, record, database);
+export const recordFactory = (record, collectionName, store) => {
+  record.save = function() {
```
Comment (zzarcon) [inline src/recordFactory.js]: Why not arrow function here?

File: src/database.js (hunk 1 above)
Comment (oskarcieslik) [inline src/database.js]: Right! :)

File: src/recordFactory.js (hunk 2 above)
Comment (oskarcieslik) [inline src/recordFactory.js]: because save() method must be scoped to the object that it's invoked on, arrow function is always bind to original record ;)

**Comment by oskarcieslik** [comment]
I will add findById after this is merged :) it's just a wrapper for `find(collectionName, { id })`;

**Pull request merged**